### PR TITLE
Partial reads

### PR DESCRIPTION
**Pull request by tstruk**
This adds partial reads support for TCTI-device

### CODE REVIEW

File: src/tss2-tcti/tcti-device.c [hunk 1]
```diff
@@ -144,19 +149,84 @@ tcti_device_receive (
...
+        LOG_DEBUG("Partial read - reading response size");
+        fds.fd = tcti_dev->fd;
+        fds.events = POLLIN;
+
+        rc_poll = poll(&fds, nfds, timeout);
+        if (rc_poll < 0) {
+            LOG_ERROR ("Failed to poll for response from fd %d, got errno %d: %s",
+                       tcti_dev->fd, errno, strerror (errno));
+        }
+        if (rc_poll < 0) {
```
Comment (AndreasFuchsSIT) [inline src/tss2-tcti/tcti-device.c]: This looks like duplicate code...

Comment (tstruk) [reply src/tss2-tcti/tcti-device.c]: Yes, you are right. I'll remove the duplicate.

**Comment by tstruk** [comment]
@AndreasFuchsSIT in order to test it change this patch is required on the driver side:
https://patchwork.kernel.org/patch/10689297/
Looks like it will be included in the 4.21 kernel.

**Comment by AndreasFuchsSIT** [comment]
@tstruk Then I propose to wait for 4.21 so we know for sure we're compatible with that kernel release...
Ok ?

**Comment by tstruk** [comment]
I'll merge it after the driver patch is merged into the kernel tree.
We don't need to check until 4.21 is out. This is disabled by default anyway and should be enabled by people who will package it.

File: src/tss2-tcti/tcti-device.c (hunk 1 above)
Comment (tstruk) [reply src/tss2-tcti/tcti-device.c]: Duplicate code removed.

**Pull request merged**